### ISSUE TITLE
Bug or weird change: nsfw extensions no longer visible with "enable nsfw on supported extensions" enabled

### ISSUE THREAD

**Issue by jeyjai** [recloudstream/cloudstream#1149]
### Steps to reproduce

1. Turn on "enable nsfw on supported extensions"
2. Open a repo with nsfw extensions

### Expected behavior

To have them visible

### Actual behavior

They're gone

### Cloudstream version and commit hash

4.3.2 b9746c2

### Android version

Android 14

### Other details

The "Enable NSFW on supported Extensions" used to disable NSFW content on extensions with regular and NSFW content, making it possible to have NSFW extensions and still not have NSFW content on extensions with regular movies and series. 
And if you didn't want to see the NSFW extensions at all, just open providers->preferred media, and disable NSFW.

But now it's all or nothing, if you want to have the NSFW extensions visible you're also forced to have NSFW covers in your search results on sorastream.

Made more sense having it the way it used to be

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to pre-release version **[Latest](https://github.com/recloudstream/cloudstream/releases)**.
- [X] If related to a provider, I have checked the site and it works, but not the app.
- [X] I will fill out all of the requested information in this form.

**Comment by Blatzar** [comment]
I think it was caused by this: https://github.com/recloudstream/cloudstream/pull/1117

**Comment by Blatzar** [comment]
I agree, it should be filtered by tvtypes not enableAdult

**Comment by JimZiii** [comment]
> > Ability to hide all the extensions(with only NSFW content) in repositories because many have cringe names, and can't even open repositories in front of anyone. So there is a way we can hide them **without bloating settings**, and that when the Providers>Enable NSFW on supported Extensions is enabled, it will hide them
> 
> This was requested

But it already existed a way of hiding them with TvTypes, and the title of the switch "enable nsfw ON SUPPORTED EXTENSIONS" implies you're activating/deactivating it on normal extensions that also have nsfw content.
the enableAdult switch didn't have any effect at all on the nsfw extensions content or even visibility in the past, only regular ones.

So the old way gave you the option to hide nsfw extensions and nsfw content on normal extensions separately, which is now impossible.
Basically making the nsfw in TvTypes completely useless since enableAdult now enabled/disables both of them at the same time

So it sounds like it was changed based on a misunderstanding

**Comment by JimZiii** [comment]
> Then have to add new settings or remove it completely But they don't want settings

No new setting needed, both settings from how it used to work are still there.

1. If you open providers->preferred content and disable nsfw, you'll hide all the nsfw extensions, that still works that way. 

2. If you disable the enableAdult switch you'll hide nsfw content in normal extensions that also have nsfw content. 

The only thing that's changed from before is that enableAdult now handles both things at the same time while both options are still available. 
All you have to do is revert the enableAdult switch back to only hiding nsfw content in normal extensions and the nsfw in preferred content well still handle the hiding of nsfw extensions

**Comment by Blatzar** [comment]
> This was requested

Yes, and it is a good feature, which was merged. But @JimZiii presents a good usecase for removing NSFW using preferred TV types over the enable nsfw checkbox. The feature will still exist, but it will not negatively impact workflows.

**Comment by Blatzar** [comment]
> How without new settings?

You know the preferred TV Types you can select, e.g TV, Anime, Movies, NSFW, if you deselect NSFW there then that will also hide it within repositories.

**Comment by D0ws1ng** [comment]
Guys just go to (providers ->preferred media ->enable NSFW)
